src/commands/watch.rs

```diff
@@ -0,0 +1,304 @@
+use std::{convert::Infallible, ffi::OsString, path::PathBuf, sync::Arc, time::Duration};
+
+use anyhow::Result;
+use clap::Parser;
+use spin_loader::local::{
+    config::{
+        RawAppManifestAnyVersion, RawAppManifestAnyVersionImpl, RawFileMount, RawModuleSource,
+    },
+    parent_dir,
+};
+use watchexec::{
+    action::{Action, Outcome},
+    config::{InitConfig, RuntimeConfig},
+    error::RuntimeError,
+    event::{Event, Priority},
+    handler::SyncFnHandler,
+    signal::source::MainSignal::Interrupt,
+    ErrorHook, Watchexec,
+};
+
+use crate::{
+    opts::{
+        APP_MANIFEST_FILE_OPT, DEFAULT_MANIFEST_FILE, WATCH_CLEAR_OPT, WATCH_DEBOUNCE_OPT,
+        WATCH_SKIP_BUILD_OPT, WATCH_WATCH_ASSETS_OPT,
+    },
+    watch_filter::{WatchFilter, DEFAULT_IGNORE_PATTERNS},
+};
+
+// TODO: Add a --component <COMPONENT> flag
+// TODO: Add a --watch <WATCH_CONFIG> flag
+// TODO: Add a --ignore <IGNORE_CONFIG> flag
+
+/// Execute spin build and spin up when watched files change.
+#[derive(Parser, Debug)]
+#[clap(
+    about = "Rebuild and restart the Spin application when files changes",
+    allow_hyphen_values = true
+)]
+pub struct WatchCommand {
+    /// Path to spin.toml.
+    #[clap(
+            name = APP_MANIFEST_FILE_OPT,
+            short = 'f',
+            long = "file",
+            default_value = DEFAULT_MANIFEST_FILE,
+        )]
+    pub app: PathBuf,
+
+    /// Clear the screen before each run.
+    #[clap(
+            name = WATCH_CLEAR_OPT,
+            short = 'c',
+            long = "clear",
+    )]
+    pub clear: bool,
+
+    /// Set the timeout between detected change and re-execution.
+    #[clap(
+        name = WATCH_DEBOUNCE_OPT,
+        short = 'd',
+        long = "debounce",
+        default_value = "100"
+    )]
+    pub debounce: u64,
+
+    /// Skip running spin build and only run spin up.
+    #[clap(name = WATCH_SKIP_BUILD_OPT, long = "skip-build")]
```

Comment:
nit: you can leave off the `long` name if it matches the field name (with conversion to kebab-case).
```suggestion
    #[clap(name = WATCH_SKIP_BUILD_OPT, long)]
```

Reply:
Cool, didn't know that. I tried it out and it made the flag names all caps which I didn't like so I'm going to keep this as is.